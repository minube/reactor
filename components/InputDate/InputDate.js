import {
  arrayOf, bool, func, oneOfType, shape, string,
} from 'prop-types';
import React, { PureComponent } from 'react';
import { View } from 'react-native';

import { LAYOUT } from '../../common';
import Calendar from '../Calendar';
import { LOCALE } from '../Calendar/modules';
import Icon from '../Icon';
import { InputHint, InputLabel } from '../Input';
import Motion from '../Motion';
import Text from '../Text';
import Touchable from '../Touchable';
import styles from './InputDate.style';
import verboseValue from './modules/verboseValue';

class InputDate extends PureComponent {
  static propTypes = {
    disabled: bool,
    error: oneOfType([bool, string]),
    hint: string,
    label: string,
    locale: shape(),
    onChange: func,
    placeholder: string,
    range: bool,
    value: oneOfType([shape(), arrayOf(shape())]),
  };

  static defaultProps = {
    disabled: false,
    error: undefined,
    hint: undefined,
    label: undefined,
    locale: LOCALE,
    onChange() {},
    placeholder: undefined,
    range: false,
    value: undefined,
  };

  constructor(props) {
    super(props);
    this.state = {
      active: false,
      calendar: props.value,
      clicks: 0,
    };
  }

  componentWillReceiveProps({ value }) {
    this.setState({ calendar: value });
  }

  _onSelect = (value) => {
    const { _onToggle, props: { onChange, range }, state: { clicks } } = this;

    this.setState({ calendar: value, clicks: clicks + 1 });
    if (!range || (value[0] && value[1])) {
      if (!range) _onToggle();
      onChange(value);
    }
  }

  _onToggle = () => {
    const { state: { active } } = this;

    this.setState({ active: !active });
  }

  _onToggle = (event) => {
    const { state: { active }, _onToggleOutside } = this;
    if (event) this.touchable = event.currentTarget.parentElement;

    if (!active) document.addEventListener('click', _onToggleOutside, false);
    else document.removeEventListener('click', _onToggleOutside, false);

    this.setState({ active: !active });
  }

  _onToggleOutside = (e) => {
    const { _onToggle } = this;
    if (this.touchable.contains(e.target)) return;
    _onToggle();
  }

  render() {
    const {
      _onSelect, _onToggle,
      props: {
        disabled, error, hint, label, locale, onChange, placeholder, range, value, ...inherit
      },
      state: { active, calendar, clicks },
    } = this;
    const { VIEWPORT: { REGULAR, LARGE } } = LAYOUT;

    return (
      <View style={[styles.container, active && styles.active, inherit.style]}>
        { label && <InputLabel>{label}</InputLabel> }

        <Touchable onPress={!disabled ? event => _onToggle(event) : undefined}>
          <View
            style={[
              styles.border,
              disabled && styles.disabled,
              !disabled && error && styles.error,
              active && styles.focus,
            ]}
          >
            <Text input lighten={disabled || !value} numberOfLines={1} style={styles.value}>
              {verboseValue(value, locale) || placeholder}
            </Text>

            { !disabled && !error && (
              <Motion timeline={[{ property: 'rotate', value: active ? '180deg' : '0deg' }]}>
                <Icon value="navDown" style={[styles.icon, styles.iconNav]} />
              </Motion>
            )}

            { !disabled && error && <Icon value="error" style={styles.icon} /> }
          </View>
        </Touchable>

        { hint && <InputHint>{hint}</InputHint> }

        { active && (
          <Calendar
            box={false}
            disabledPast={inherit.disabledPast}
            edges={inherit.edges}
            expanded={REGULAR || LARGE}
            locale={locale}
            range={range}
            onSelect={_onSelect}
            style={[styles.calendar, label && styles.withLabel]}
            value={calendar}
            clicks={clicks}
          />
        )}
      </View>
    );
  }
}

export default InputDate;
