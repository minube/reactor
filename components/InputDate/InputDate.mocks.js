import { TOMORROW, NEXT_MONTH } from '../Calendar/Calendar.mocks';


const ERROR = 'Something happened...';

const HINT = 'Optional field';

const LABEL = 'Date';

const STYLE = {
  backgroundColor: 'rgba(0,255,0,0.25)', paddingBottom: 32, marginRight: 16, width: 256,
};

const VALUE = TOMORROW;
const VALUE_RANGE = [TOMORROW, NEXT_MONTH];

export {
  ERROR,
  HINT,
  LABEL,
  STYLE,
  VALUE,
  VALUE_RANGE,
};
