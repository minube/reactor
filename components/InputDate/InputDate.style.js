import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';
import { INPUT_HEIGHT } from '../Input/Input.style';

const {
  BORDER_RADIUS, COLOR, ELEVATION, SPACE,
} = THEME;

export default StyleSheet.create({
  active: {
    zIndex: 1,
  },

  border: {
    ...LAYOUT.STYLE.ROW,
    backgroundColor: COLOR.BACKGROUND_INPUT,
    borderColor: COLOR.BASE,
    borderRadius: BORDER_RADIUS,
    borderWidth: 1,
    minHeight: INPUT_HEIGHT,
    paddingHorizontal: SPACE.XXS,
    width: '100%',
  },

  calendar: {
    ...ELEVATION.REGULAR,
    paddingTop: SPACE.S,
    position: 'absolute',
    top: INPUT_HEIGHT,
    zIndex: 1,
  },

  container: {
    marginBottom: SPACE.REGULAR,
    maxWidth: '100%',
  },

  disabled: {
    backgroundColor: COLOR.DISABLED,
  },

  error: {
    borderColor: COLOR.ERROR,
  },

  focus: {
    borderColor: COLOR.PRIMARY,
  },

  icon: {
    marginHorizontal: SPACE.XXS,
  },

  iconNav: {
    height: SPACE.MEDIUM,
  },

  value: {
    flex: 1,
    paddingHorizontal: SPACE.XXS,
  },

  withLabel: {
    marginTop: SPACE.REGULAR + SPACE.XXS,
  },
});
