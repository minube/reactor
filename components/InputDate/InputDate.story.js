import React, { Component } from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import {
  text, boolean, number, object, select,
} from '@storybook/addon-knobs/react';

import InputDate from './InputDate';
import {
  ERROR, HINT, LABEL, STYLE, TEMPLATE, VALUE, VALUE_RANGE
} from './InputDate.mocks';

class HOC extends Component {
  constructor(props) {
    super(props);
    this.state = { value: props.value };
  }

  componentWillReceiveProps({ value }) {
    this.setState({ value });
  }

  _onChange = (value) => {
    const { props: { onChange } } = this;

    this.setState({ value });
    if (onChange) onChange(value);
  }

  render() {
    const { _onChange, props, state: { value } } = this;

    return <InputDate {...props} value={value} onChange={_onChange} />;
  }
}

storiesOf('✅ InputDate', module)
  .add('default', () => (
    <HOC />
  ))
  .add('disabled', () => (
    <HOC disabled />
  ))
  .add('error', () => (
    <HOC error={ERROR} />
  ))
  .add('hint', () => (
    <HOC hint={HINT} />
  ))
  .add('label', () => (
    <HOC label={LABEL} />
  ))
  .add('value', () => (
    <HOC value={VALUE} />
  ))
  .add('value & range', () => (
    <HOC range value={VALUE_RANGE} />
  ))
  .add('⚡ onChange', () => (
    <HOC onChange={action('InputDate.onChange()')} title="Press me" />
  ))
  .add('style', () => (
    <HOC style={STYLE} />
  ))
  .add('🏀 Playground', () => (
    <HOC
      disabled={boolean('disabled', false)}
      error={text('error', null)}
      hint={text('hint', HINT)}
      label={text('label', LABEL)}
      value={object('value', VALUE_RANGE)}
      range={boolean('range', true)}
      style={object('style', STYLE)}
    />
  ));
