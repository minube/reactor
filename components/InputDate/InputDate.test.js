import React from 'react';
import renderer from 'react-test-renderer';

import InputDate from './InputDate';
import {
  ERROR, HINT, LABEL, STYLE, TEMPLATE, VALUE, VALUE_RANGE
} from './InputDate.mocks';

describe('<InputDate>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputDate />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputDate disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {error}', () => {
    const tree = renderer.create(<InputDate error={ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputDate hint={HINT} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputDate label={LABEL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputDate value={3} value={VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value} & {range}', () => {
    const tree = renderer.create(<InputDate value={3} value={VALUE_RANGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputDate style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
