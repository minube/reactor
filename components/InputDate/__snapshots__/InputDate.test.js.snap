// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDate> renders 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="none"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            Object {
              "color": "#707070",
            },
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {error} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          Object {
            "borderColor": "#ef5160",
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/error.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "marginHorizontal": 4,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {hint} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 12,
          "lineHeight": 16,
        },
        Object {
          "color": "#707070",
        },
        Object {
          "marginTop": 4,
          "maxWidth": "100%",
          "paddingLeft": 0,
        },
      ]
    }
  >
    Optional field
  </Text>
</View>
`;

exports[`<InputDate> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontFamily": undefined,
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 16,
        },
        Object {
          "color": "#707070",
        },
        Object {
          "color": "#4a4a4a",
          "marginBottom": 8,
        },
      ]
    }
  >
    Date
  </Text>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "marginRight": 16,
        "paddingBottom": 32,
        "width": 256,
      },
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {value} & {range} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        3 jun. 2019 - 1 jul. 2019
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputDate> when {value} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
        "maxWidth": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 3,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "minHeight": 42,
            "paddingHorizontal": 4,
            "width": "100%",
          },
          false,
          undefined,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            false,
            Object {
              "flex": 1,
              "paddingHorizontal": 8,
            },
          ]
        }
      >
        3 jun. 2019
      </Text>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg0",
              },
            ],
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/nav-arrow.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Array [
                Object {
                  "marginHorizontal": 4,
                },
                Object {
                  "height": 16,
                },
              ],
              Object {
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 3,
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#cdd2d4",
            "height": 0,
            "opacity": 0,
            "width": 0,
            "zIndex": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
